spec/unit/typesafe/config/path_spec.rb

```diff
@@ -0,0 +1,259 @@
+      expect("Path(foo.bar)").to eq(TestUtils.path("foo", "bar").to_s)
+      expect('Path(foo."bar*")').to eq(TestUtils.path("foo", "bar*").to_s)
+      expect('Path("foo.bar")').to eq(TestUtils.path("foo.bar").to_s)
+    end
+  end
+
+  ####################
+  # Render
+  ####################
+  context "testing .render" do
+    context "rendering simple one element case" do
+      let(:expected) { "foo" }
+      let(:path) { TestUtils.path("foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering simple two element case" do
+      let(:expected) { "foo.bar" }
+      let(:path) { TestUtils.path("foo", "bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering non safe char in an element" do
+      let(:expected) { 'foo."bar*"' }
+      let(:path) { TestUtils.path("foo", "bar*") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering period in an element" do
+      let(:expected) { '"foo.bar"' }
+      let(:path) { TestUtils.path("foo.bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering hyphen in element" do
+      let(:expected) { "foo-bar" }
+      let(:path) { TestUtils.path("foo-bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering hyphen in element" do
+      let(:expected) { "foo_bar" }
+      let(:path) { TestUtils.path("foo_bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element starting with a hyphen" do
+      let(:expected) { "-foo" }
+      let(:path) { TestUtils.path("-foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element starting with a number" do
+      let(:expected) { "10foo" }
+      let(:path) { TestUtils.path("10foo") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering empty elements" do
+      let(:expected) { '"".""' }
+      let(:path) { TestUtils.path("", "") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering element with internal space" do
+      let(:expected) { '"foo bar"' }
+      let(:path) { TestUtils.path("foo bar") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering leading and trailing spaces" do
+      let(:expected) { '" foo "' }
+      let(:path) { TestUtils.path(" foo ") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering trailing space only" do
+      let(:expected) { '"foo "' }
+      let(:path) { TestUtils.path("foo ") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering number with decimal point" do
+      let(:expected) { "1.2" }
+      let(:path) { TestUtils.path("1", "2") }
+      include_examples "path_render_test"
+    end
+
+    context "rendering number with multiple decimal points" do
+      let(:expected) { "1.2.3.4" }
+      let(:path) { TestUtils.path("1", "2", "3", "4") }
+      include_examples "path_render_test"
+    end
+  end
```

Comment:
rspec also makes it awkward to extract tests into helper methods. Calling `expect` from inside a helper method throws an error because `expect` statements have to appear in an `it` or `specify` block as near as I can tell. Either that or I can't rspec very well. Agreed that it looks weird like this, but like @fpringvaldsen said, it seems to be the convention